endpoints/data_objects/src/main.cpp

```diff
@@ -1369,12 +1371,35 @@ namespace
 			res.keep_alive(_req.keep_alive());
 
 			try {
+				DataObjInp input{};
+				irods::at_scope_exit free_memory{[&input] { clearKeyVal(&input.condInput); }};
+
+				input.oprType = UNREG_OPR;
+
 				const auto lpath_iter = _args.find("lpath");
 				if (lpath_iter == std::end(_args)) {
 					log::error("{}: Missing [lpath] parameter.", fn);
 					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
 				}
 
+				std::strncpy(input.objPath, lpath_iter->second.c_str(), sizeof(DataObjInp::objPath));
+				addKeyVal(&input.condInput, COPIES_KW, "0");
+
+				if (const auto iter = _args.find("resource"); iter != std::end(_args)) {
+					addKeyVal(&input.condInput, RESC_NAME_KW, iter->second.c_str());
+				}
+				else if (const auto iter = _args.find("replica-number"); iter != std::end(_args)) {
+					addKeyVal(&input.condInput, REPL_NUM_KW, iter->second.c_str());
+				}
+				else {
+					log::error("{}: Missing [resource] or [replica-number] parameter.", fn);
```

Comment:
ah, so one of the two *is* required to be present... got it... we should say that in the help text / docs.

Reply:
Done